src/components/shared/pickers/DateTimePickerCTA.tsx

```diff
@@ -32,7 +34,22 @@ const formatDate = (formatValue: Date) => {
 };
 
 export function CustomLayout(props: PickersLayoutProps<any, any, any>) {
+    const intl = useIntl();
     const { shortcuts, toolbar, tabs, content } = usePickerLayout(props);
+
+    let currentStep = '';
+    if (
+        content &&
+        typeof content === 'object' &&
+        // @ts-expect-error How we call this we should get a react node
+        content.props &&
+        // @ts-expect-error We check if the thing exists so we should be okay
+        typeof content.props === 'object'
+    ) {
+        // @ts-expect-error being pretty safe up above
+        currentStep = content.props.focusedView ?? content.props.view ?? '';
+    }
```

Comment:
How would you feel about replacing the `if` condition with a custom type guard that evaluates whether a `ReactNode` is a `ReactElement`?

The type guard would take the following shape and stored off as a util:
```ts
export const isReactElement = (value: ReactNode): value is ReactElement =>
    isObject(value) && 'props' in value;
```